libs/nss.c

```diff
@@ -108,8 +122,15 @@ int _nss_tcb_getspent_r(struct spwd *__result_buf,
 
 	do {
 		currpos = telldir(tcbdir);
+#if USE_READDIR
+		saved_errno = errno;
```

Comment:
Looks like `errno = saved_errno;` is needed for `result != NULL` case if `USE_READDIR` because, if I'm not mistaken, `readdir` is allowed to clobber `errno`.

Reply:
In the case of `(!result && errno)` `errno` is restored in line 135.  On `(!result)` `errno` will be set to `ENOENT`.  That's how the logic works with `!USE_READDIR` also.

Reply:
`(!result)` coveres the case of an empty dir if I got the logic behind the code correctly.

Reply:
I mean `result != NULL` case.

Reply:
Yes, now.  It's pretty late over here.  Will fix that.

Reply:
Fix is now present on line 148 to 150.